### ISSUE TITLE
Custom element not upgraded when created using "innerHTML" on some platforms

### ISSUE THREAD

**Issue by argelius** [WebReflection/document-register-element#75]
Hello @WebReflection !

Thanks for the polyfill. It's great! :)
We use it for our hybrid app library, Onsen UI: https://github.com/OnsenUI/OnsenUI

I've noticed an issue when elements are created by modifying the `innerHTML` property of an element. I think it's easiest to show by an example.

The `<ons-alert-dialog>` element in the following example has a method called `show` but as you can see the element is only upgraded when using `document.createElement` and not by adding it through `innerHTML`:

```
var a = document.createElement('ons-alert-dialog');

alert(a.show); // This works on all platforms and correctly shows the function.

var b = document.createElement('div');
b.innerHTML = '<ons-alert-dialog></ons-alert-dialog>';
b = b.firstChild;

alert(b.show); // This shows "undefined" on some platforms
```

I've noticed this in Firefox 48 and older Android browsers such as Android 4.2.2. Works fine in Chrome and Safari.

**Comment by WebReflection** [comment]
One of the caveats that this poly carries since everywhere to avoid impacting performance too much:
https://github.com/WebReflection/document-register-element#using-innerhtml

TL;DR the helper gives you `innerHTML(el, html)`

If you want to trigger this polyfill (maybe with a trotled logic) you need to update `HTMLElement.prototype.innerHTML` and use original getter and trigger updates on setter.

I haven't done it here and not sure I should.

**Comment by argelius** [comment]
@WebReflection Thanks! Looks like I should have read the README more carefully. Sorry about that :)

**Comment by WebReflection** [comment]
do you think we can close this?

**Comment by argelius** [comment]
@WebReflection Sorry for the late reply. I think `innerHTML` helper is sufficient.
